### DISCUSSION TITLE
Tự học lập trình đối với kỹ sư trái ngành

### DISCUSSION THREAD

**Discussion by huypham0808** [chuongmep/blogcomment#7] [Q&A]
Chào anh Chương, em tên Huy, em là kỹ sư xây dựng tự học lập trình. Em có theo dõi anh trên facebook và kênh youtube. Được biết anh cũng là một người tự học lập trình và đã đi theo con đường R&D như một dev chuyên nghiệp ở Singapore. Hiện em cũng mong muốn đi theo con đường này trong tương lai. Em xin tự nói về bản thân 1 chút với mong muốn nhận được những chia sẽ của anh cho việc định hướng học tập trong tương lai. 

Em  học về kỹ thuật xây dựng nên từ khi còn học đại học em đã có tư duy về việc tự động hoá các bản tính toán các cấu kiện. Khi đi làm thì em có tự học thêm về C# để viết các tool nho nhỏ trong Autocad để hỗ trợ công việc cũng như những tool về desktop app (ứng dụng window form, WPF). Hiện tại em đang học C++, học thêm về Data Structure and algorithms để xây dựng base cơ bản vì là tay ngang, không có kiến thức nền như sinh viên CNTT, em cũng tự học thêm về OOP và áp dụng các Design Pattern (MVVM) vào các tool mình đang viết.  Học các kiến thức cơ bản về Html, CSS, Javascript để có thể tương lai sẽ làm về Wep App. 

Là một người đi trước anh có thể chia sẽ về lộ trình, cũng như kiến thức anh đã tự học như thế nào để có được một outcome của một dev không ạ. Em cảm ơn.

**Comment by chuongmep** [comment]
Chào @Huydev0808 , lộ trình cá nhân không phải do người khác quyết định mà chính ở mỗi cá nhân. Khi nhìn thấy điểm mạnh của cá , hãy theo đuổi vào điểm mạnh của mình và xây dựng nó đến cấp độ cao nhất. Khi chưa tìm ra điểm mạnh, hãy tiếp tục khám phá thế giới xung quanh mình. Việc học là suốt đời, nền tảng là cốt lõi không phải gì cũng học một ít để biết rồi bỏ ngang. Em học nhiều thứ nhưng em đã thành công cho bao nhiêu ý tưởng và dự án thực tế cho những thứ mình đã học , hãy suy nghĩ về điều đó. Chúc em thành công.
